src/pages/films.js

```diff
@@ -1,54 +1,113 @@
-import React, {Fragment} from "react";
-import { graphql } from "gatsby";
-
-
-export default function Films({ data }) {
-   
-    const films = data.hasura.tnff_films
-    // filter film by year
-    // gallery with film image and title
-    const filmData = films.map(film => {
-        const { film_img_url, title, id, release_date } = film;
-        if (film_img_url) {
-            return (
-                <div className="film_single" key={id}>
-                    <img src={`${film_img_url}`} alt="" />
-                    <h2>{title}</h2>
-                </div>
-            );
-        } else {
-            return null
-        }
-    })
-    return (
-        <div className="films_container">
-            {/* <pre>{ JSON.stringify(data.hasura, null, 2) }</pre> */}
-            {filmData}
-        </div>
-    )
-}
+import React, { Fragment, useState, useCallback, useEffect } from "react";
+import Layout from '../components/layout'
+
+
+export default function Films({ }) {
+
+  const [data, setData] = useState([]);
+  const [filmFilter, setFilmFilter] = useState("all");
+  const [films, setFilms] = useState([]);
+  const [searchYear, setSearchYear] = useState('')
 
-export const query = graphql`
-    query FilmQuery{
-        hasura {
-            tnff_films {
-                id
-                title
-                tnff_year
-                location
-                category
-                release_date
-                duration_mins
-                language
-                director
-                producer
-                synopsis
-                director_bio
-                film_img_url
-                director_img_url
-                other_info
-            }
-        }
+
+  // ! breakdown tasks/lil sprints for timeline
+
+  const getData = useCallback(async function () {
+    try {
+      const data = await fetch("https://sonal-hasura.herokuapp.com/v1/graphql", {
+        "method": "POST",
+        "headers": {
+          "Content-Type": "application/json"
+        },
+        "body": "{\"query\":\"{\\n tnff_films (order_by: {tnff_year: desc}) {\\n                id\\n                title\\n                tnff_year\\n                location\\n                category\\n                release_date\\n                duration_mins\\n                language\\n                director\\n                producer\\n                synopsis\\n                director_bio\\n                film_img_url\\n                director_img_url\\n                other_info\\n            }\\n}\"}"
+      });
```

Comment:
same here, the formatting

Reply:
removed allfilms.js, was a failed helper file